### DISCUSSION TITLE
Hot to interact with DOM?

### DISCUSSION THREAD

**Discussion by krm441** [emscripten-core/emscripten#17588] [Show and tell]
I have read all the possible tutorials and docs. But i still don't understand how i can use a simple html button to invoke some changes in my c++. For example:
`<div><button onclick="jump()"> Button </button></div>`

in C++;
`inline void jump()
{
     Player::GetInstance()->jump();
 }
`
How can I do that?

Please forgive me; emscripten is a new area for me. I use glfw and on desktop it perfectly responds to all keys. But I need to create a GUI. So i thought that using html/css buttons will be a good idea.

**Comment by sbc100** [comment]
The simplest way to make the work would be to export the `jump` function.   Do do that you can either tag it with `EMSCRIPTEN_KEEPALIVE` or add it to `-sEXPORTED_FUNCTIONS` on the command line.

Also if you are writing it in C++ you will need to mark it as `extern "C"`.

Finally when can call a native function from JS you need to prefix it with a leading underscore.  So that call would be `_jump()` or `Module._jump()`

**Reply by krm441** [reply]
Thank you! Everything works.
